content/applications/inventory_and_mrp/inventory/barcode/operations/transfers_scratch.rst

```diff
@@ -1,17 +1,200 @@
+   the :guilabel:`pencil icon` to manually enter the transferred quantities.
+
+   .. image:: transfers_scratch/transfers-scratch-product-line-editor.png
+      :align: center
+      :alt: Product line editor for individual transfer in Barcode app.
+
+Additionally, the :guilabel:`+1` and :guilabel:`-1` buttons can be clicked to add or subtract
+quantity of the product, and the :guilabel:`number keys` can be used to add quantity, as well.
+
+Below the :guilabel:`number keys` are the two :guilabel:`location` lines, which read `WH/Stock` by
+default, or whichever locations were previously specified on the internal transfer form. Click these
+lines to reveal a drop-down menu of additional locations to choose from.
+
+Once ready, click :guilabel:`Confirm` to confirm the changes made to the product line.
+
+Then, from the overview page with all products to process within that transfer (**WH/INT/000XX**),
+click :guilabel:`Validate`. The receipt has now been processed, and the :guilabel:`Barcode app` can
+be closed out.
+
+.. tip::
+   The *Barcode* app can also be used to scan products in internal transfers containing unique lot
+   numbers and serial numbers. From the barcode transfer screen, scan the barcode of a lot or serial
+   number, and Odoo will automatically increase the quantity of the product to the quantity recorded
+   in the database. If the same lot or serial number is shared between different products, scan the
+   product barcode first, then the barcode of the lot/serial number.
+
+Create a transfer from scratch
 ==============================
-Create a Transfer from Scratch
-==============================
 
-To create a transfer from the *Barcode* application, you first need to
-print the operation type barcodes. To do so, you can download the
-*Stock barcode sheet* from the home page of the app.
+In addition to processing and scanning barcodes for existing, previously-created internal transfers,
+the :guilabel:`Barcode app` can also be used to create transfers from scratch, simply by scanning a
+printed operation type barcode.
+
+.. admonition:: Did you know?
+
+   Odoo's *Barcode* application provides demo data with barcodes to explore the features of the app.
+   These can be used for testing purposes, and can be printed from the home screen of the app. To
+   access this demo data, navigate to :menuselection:`Barcode app` and click :guilabel:`stock
+   barcodes sheet` (bolded and highlighted in blue) in the information pop-up above the scanner.
+
+   .. image:: transfers_scratch/transfers-scratch-demo-data.png
+      :align: center
+      :alt: Demo data prompt pop-up on Barcode app main screen.
+
+To do this, first navigate to the :menuselection:`Barcode app`. Once inside the :guilabel:`Barcode
+app`, a :guilabel:`Barcode Scanning` screen displaying different options is presented. From this
+screen, click the :guilabel:`Tap to Scan` button (next to the camera icon, at the center of the
```

Comment:
*Warning*: I'm getting very nitpicky here :sweat_smile: 

Your current instruction is correct when the barcode scanner is the mobile phone (:
Wanted to add to this sentence that when we use a USB barcode scanner that's connected to the Odoo Barcode mobile app, the actual scanning is performed by the USB barcode scanner (like a zebra scanner)

```suggestion
screen, when using a USB or bluetooth barcode scanner, directly scan the product barcode. When using a smartphone as the barcode scanner, click the :guilabel:`Tap to Scan` button (next to the camera icon, at the center of the
```

If my suggestion is accepted, I'd also link to the doc about the [barcode scanner types](https://www.odoo.com/documentation/16.0/applications/inventory_and_mrp/barcode/setup/hardware.html) (also I'm improving this doc! :smile: ):

Reply:
Not nitpicky, that's an important point to add! And for linking to that page, I already did so at the top of the page. Do you think I should do it here again? And glad it's getting updated!